tools/src/main/java/org/apache/kafka/tools/ConsumerPerformance.java

```diff
@@ -295,16 +287,6 @@ public ConsumerPerfOptions(String[] args) {
                 .describedAs("size")
                 .ofType(Integer.class)
                 .defaultsTo(2 * 1024 * 1024);
-            numThreadsOpt = parser.accepts("threads", "DEPRECATED AND IGNORED: Number of processing threads.")
```

Comment:
Please update e2e file https://github.com/apache/kafka/blob/b9fe9f532fe11c658d07b11543e874526cb837b8/tests/kafkatest/services/performance/consumer_performance.py#L124

DON'T set the removed config in the e2e when the version >= 4.0.